src/Compilers/Core/Portable/Operations/IOperation.Enumerable.cs

```diff
@@ -13,15 +13,15 @@
 
 namespace Microsoft.CodeAnalysis
 {
-    internal abstract partial class Operation : IOperation
+    public partial interface IOperation
     {
         /// <summary>
         /// Implements a struct-based enumerable for <see cref="Operation"/> nodes, using a slot-based system that tracks
         /// the current slot, and the current index in the slot if the current slot is an immutable array. This type is not hardened
         /// to <code>default(Enumerable)</code>, and will null reference in these cases.
         /// </summary>
         [NonDefaultable]
-        internal readonly struct Enumerable : IEnumerable<IOperation>
+        public readonly struct Enumerable : IEnumerable<IOperation>
```

Comment:
❔ Do value types nested in an interface cause problems for any common languages/versions? If I try to use it in a normal net472 app, it gives CS8370 about feature 'default interface implementation'. However, if I set `LangVersion` to 9 it does appear to execute without a runtime error.

Reply:
I did not realize that you couldn't do this before... I'll refactor this into `IOperationEnumerable`, I guess.

Reply:
I would not object to treating this as a point-in-time problem provided the following were met:

1. The framework team feels that this pattern reflects a long term "recommended design" for similar cases
2. There are no known cases that we need to support where this pattern causes runtime failures (complete inability to execute or use the feature)

Reply:
I don't think we want to recommend our consumers put themselves into an explicitly unsupported configuration.